src/lastAddedBlock.ts

```diff
@@ -33,26 +37,33 @@ export class LastAddedBlock {
         blockClickedSubject: Subject<SkipBlock>
     ) {
         this.flash = flash;
+        this.roster = roster;
+        this.initialBlock = initialBlock;
+        this.blockClickedSubject = blockClickedSubject;
 
         // Main SVG canvas that contains the last added block of the chain
-        const svgLast = d3
+        this.svgLast = d3
             .select("#last-container")
             .attr("height", this.svgHeight);
 
         // Fetch the last block from the Cothority client
-        new SkipchainRPC(roster)
-            .getLatestBlock(initialBlock.hash, false, true)
+        this.getLastBlock();
+    }
+
+    private async getLastBlock() {
```

Comment:
instead of changing `roster`, `initialBlock`, and `blockClickedSubject` as attributes, why not just providing them as arguments to `getLastBlock` ?

Reply:
I forgot to mention `svgLast`  too 😁